src/components/TodoItem.jsx

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable react/prop-types */
+import { useState, useRef } from 'react';
+import { FaTrash } from 'react-icons/fa';
+import { AiFillEdit } from 'react-icons/ai';
+import { useTodosContext } from '@/context/TodosContext';
+import { useAuthContext } from '@/context/AuthContext';
+import styles from '@/styles/TodoItem.module.scss';
+
+const TodoItem = ({ itemProp }) => {
+  const [editing, setEditing] = useState(false);
+
+  const { handleChange, delTodo, setUpdate } = useTodosContext();
+  const { user } = useAuthContext();
+
+  const editInputRef = useRef(null);
+
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  const viewMode = {};
+  const editMode = {};
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  const handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      setUpdate(editInputRef.current.value, itemProp.id);
+      setEditing(false);
+    }
+  };
+  return (
+    <li className={styles.item}>
+      <div className={styles.content} style={viewMode}>
+        <input
+          type="checkbox"
+          checked={itemProp.completed}
+          onChange={() => handleChange(itemProp.id)}
+        />
+        {user && (
+          <button type="button" onClick={handleEditing}>
+            <AiFillEdit
+              style={{ color: '#5e5e5e', fontSize: '16px' }}
+            />
+          </button>
+        )}
+        <button type="button" onClick={() => delTodo(itemProp.id)}>
+          <FaTrash style={{ color: '#5e5e5e', fontSize: '16px' }} />
+        </button>
```

Comment:
[Optional] `delete' is part of edit operations, so supposed to be included in the logic;
```suggestion
        {user && (
          <button type="button" onClick={handleEditing}>
            <AiFillEdit
              style={{ color: '#5e5e5e', fontSize: '16px' }}
            />
          </button>
        )
        <button type="button" onClick={() => delTodo(itemProp.id)}>
          <FaTrash style={{ color: '#5e5e5e', fontSize: '16px' }} />
        </button>}
```

Reply:
@adamilare I think the login is meant to grant users access to the edit function alone. A task can be deleted without logging in.